updates the plugin to work with newest oh-my-fish (closes #1)

douglascamata opened the pull request:
exenv.fish renamed to init.fish and removed the init function, along with uninstall function, as they are not necessary anymore.

@derekstavis, can you do a quick review, please? Thanks.

douglascamata committed:
updates the plugin to work with newest oh-my-fish (closes #1)

exenv.fish renamed to init.fish and removed the init function, along with
uninstall function, as they are not necessary anymore.

bobthecow commented on init.fish:
```diff
@@ -0,0 +1,30 @@
+
+    set PATH "$EXENV_ROOT/shims" $PATH
+
+    function ndenv -d "Elixir version manager"
```
would it make sense to move this function into `functions/` where it can be autoloaded, reloaded when updated, etc?

douglascamata replied:
You mean when the plugin itself is updated?

bobthecow replied:
yup!

derekstavis replied:
I believe this implementation was somewhat intentional... Not defining the function if not in interactive mode and when `exenv` does not exist.

douglascamata replied:
@derekstavis I realize that now, hahaha. I'll improve the second conditional to only execute if exenv is defined.

douglascamata replied:
@bobthecow indeed I can't move this code to the `functions` folder, because it's all meant to run when the shell is loading. It doesn't feel right to put this code anywhere else.

derekstavis commented on init.fish:
```diff
@@ -0,0 +1,30 @@
+      set -e argv[1]
+      switch "$command"
+        case shell
+          eval (ndenv "sh-$command" $argv)
```
I see this `ndenv` here, I'm not a `exenv` user, but looks like the wrong command line is being called here

douglascamata replied:
I thought this was actually doing something (I forked this code from an old and not-maintained-anymore repository). But I searched in exenv's and didn't find anything about it. I'll remove it.

douglascamata committed:
removing unused code and refactoring

Removed the unused ndenv function and improved the conditional for
evenv setup.

douglascamata commented:
@derekstavis much clearer now, imho. What you think?
